projects/m5/001-budget-app/js/class.js

```diff
@@ -0,0 +1,127 @@
+class Category {
+  /**
+   * Creates a new category
+   *
+   * @param {string} category - Name of the new category to be created
+   */
+  constructor(category) {
+    this.category = category;
+    this.ledger = [];
+    this.balance = 0;
+  }
+
+  /**
+   * Deposit a certain amount of money to this category's ledger
+   *
+   * @param {number} amount - The amount of money to deposit
+   * @param {string} [description = ''] - Optional: the description of the operation
+   */
+  deposit(amount, description = "") {
+    if (this.checkAmount(amount)) {
+      this.ledger.push({ amount: +amount, description: description });
+      this.balance += +amount;
+    }
+  }
+
+  /**
+   * Withdraw a certain amount of money from the category's ledger, if there is enough deposited money
+   *
+   * @param {number} amount - The amount of money to withdraw
+   * @param {string} [description = ''] - Optional: the description of the operation
+   * @returns {boolean} - True if the operation was successful, otherwise false
+   */
+  withdraw(amount, description = "") {
+    if (this.checkFunds(amount) && this.checkAmount(amount)) {
+      this.ledger.push({
+        amount: -amount,
+        description: description,
+      });
+      this.balance -= amount;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Get the balance from this category
+   *
+   * @returns {string} - returns formatted balance with two decimals
+   */
+  getBalance() {
+    return +this.balance.toFixed(2);
```

Comment:
What would happen if I asked you for the balance relating to only the first 5 movements and **not the absolute balance**?

Reply:
I see, I actually hadn't thought about this feature. I will solve this issue!